src/smartpeak/include/SmartPeak/ui/FilePicker.h

```diff
@@ -45,19 +48,23 @@ namespace SmartPeak
     bool loading_is_done_ = true;
     bool file_was_loaded_ = true;
     bool error_loading_file_ = false;
+    std::atomic_bool files_scanned {false};
 
     void run_and_join(
       FilePickerProcessor* processor,
       const std::string& pathname,
       bool& loading_is_done,
       bool& file_was_loaded
     );
+    
+    ///!  rescan pathname_content_ into content_items when needed
+    void update_contents(ImVector<ImDirectoryEntry>& content_items);
 
   public:
     FilePicker()
     {
       // current_pathname_ = fs::current_path().root_path().string();
-      pathname_content_ = Utilities::getPathnameContent(current_pathname_);
+      // pathname_content_ = Utilities::getPathnameContent(current_pathname_);
```

Comment:
Is this needed or should it be removed?

Reply:
They are no longer needed and will be eradicated